File: herencia-dto/HerenciaDto-Java-IvanRC/src/main/java/dto/PersonaDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "persona")
@Root(name = "persona")
public class PersonaDto {
    @Json(name = "nombre")
    @Element(name = "nombre")
    String nombre;

    @Json(name = "edad")
    @Element(name = "edad", required = false)
    String edad;

    @Json(name = "modulo")
    @Element(name = "modulo", required = false)
    String modulo;

    @Json(name = "tipo")
    @Element(name = "tipo")
    String tipo;
}

File: herencia-dto/HerenciaDto-Kotlin-IvanRC/src/main/kotlin/dto/PersonaDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Json(name = "persona")
@Root(name = "persona")
data class PersonaDto(
    @Json(name = "nombre")
    @field:Element(name = "nombre")
    @param:Element(name = "nombre")
    val nombre: String,

    @Json(name = "edad")
    @field:Element(name = "edad", required = false)
    @param:Element(name = "edad", required = false)
    val edad: String? = null,

    @Json(name = "modulo")
    @field:Element(name = "modulo", required = false)
    @param:Element(name = "modulo", required = false)
    val modulo: String? = null,

    @Json(name = "tipo")
    @field:Element(name = "tipo")
    @param:Element(name = "tipo")
    val tipo: String
)